### DISCUSSION TITLE
Tensor product basis usage for spin systems

### DISCUSSION THREAD

**Discussion by adithi16** [QuSpin/QuSpin#587] [Q&A] [answered]
Hi, 
I would like to work on a Hamiltonian with two different spin basis on the same site. We have spin= 1/2 with two Ising variable $\sigma_i$ and $\tau_i$ on every site $i$. 
I have a Hamiltonian which has a coupling term: $H_c = \sum_i \sigma_i^z  \sigma_{i+1}^z  \tau_i^z \tau_{i+1}^z$ 

How do I incorporate this term in quspin? 
I started by defining two separate spin basis and the tensor product of them:

`L=4`
`s_basis= spin_basis_1d(L=L,S='1/2', pauli =True)`
`t_basis= spin_basis_1d(L=L,S='1/2', pauli =True)`
`f_basis = tensor_basis(s_basis,t_basis) `

Then I define the coupling: 

`Coupling = [[1,i,i+1] for i in range(L-1)]`
`static= [["zz|zz",Coupling]]`

I get the following error: 
number of indices doesn't match opstr in: zz|zz, [0, 1]

Kindly let me know how this can be done. It would be greatly helpful!

Thanks

**Comment by PFmix** [comment, marked as answer]
You were almost there: in your coupling table, you have to put the index for each *four* operator, even if it appears redundant. You only mention the first two.
You have to take:
```
Coupling = [[1.,i,i+1,i,i+1] for i in range(L-1)]
```

**Reply by adithi16** [reply]
Thank you so much! That works :)
